### PR TITLE
Allow using project IDs instead of names for gitlab

### PR DESCRIPTION
**Pull request by ivan-cukic**

### CODE REVIEW

File: bugwarrior/services/gitlab.py [hunk 1]
```diff
@@ -284,7 +284,9 @@ def add_default_namespace(self, repo):
         otherwise, the loggin will be prepended as namespace:
             e.g. "bar" → "<login>/bar"
         """
-        if repo.find('/') < 0:
+        if repo.startswith('id:'):
+            return repo
+        elif repo.find('/') < 0:
```
Comment (ryneeverett) [inline bugwarrior/services/gitlab.py]: Couldn't this be simplified to:

```py
if repo.find('/') < 0 and not repo.startswith('id:'):
    return self.login + '/' + repo
else:
    return repo
```

**Review by ryneeverett — COMMENTED** [review]
What's the motivation for using id rather than name?

File: bugwarrior/services/gitlab.py (hunk 1 above)
Comment (ivan-cukic) [reply bugwarrior/services/gitlab.py]: The reason why I decided to separate the conditions is that they belong to different /features/. But I have nothing against merging them.

**Comment by ivan-cukic** [comment]
IDs are more stable than names.

Also, I've encountered some peculiar gitlab setups where name-based project retrieval through the API did not work.

**Review by ryneeverett — APPROVED** [review]

**Comment by ivan-cukic** [comment]
Any news on merging this?

**Pull request merged**